### ISSUE TITLE
[Suggestion]: Routing of passengers and freight

### ISSUE THREAD

**Issue by jirkovoriginal** [JGRennison/OpenTTD-patches#462]
Hi, I have a suggestion for a possible modification to the routing of people and material in the game. 
I'll use slow passenger trains (Os) that stop everywhere and fast passenger trains (R) that only stop at important stations as an example.
I, as a person in the game, would like to get from station 7 to station 2. The first train that arrives is Os 10001, I get on and go to station 5. Normally as a human I would continue to station 2 with the same train (green line), but in the game I think that I will get there earlier with the fast train R 1003 (red line), although this is not true because the fast train does not stop at station 2 and therefore I have to change again at station 3 to the passenger train Os 10003. 
 

![gvd1](https://user-images.githubusercontent.com/109038984/208670991-ecdd554f-96e6-444b-91b0-34a04c681918.png)

The main problem is that the fast train stops at fewer stations, so the game thinks that the journey is faster or shorter. Unfortunately, this often causes traffic problems, where fast trains are overloaded due to passengers who can travel on passenger trains and passenger trains run almost empty.
The solution is actually almost in the game. There is an option in the timetable to change the stop at a station -->  through stop (to station, via station) and thus basically tell the program that I am taking the same route. If I start from station 5 and go to station 3 it doesn't make that much difference if I take a passenger train with the timetable - to station 5, to station 4, to station 3 or an fast train - to station 5, via station 4, to station 3. 

![gvd2](https://user-images.githubusercontent.com/109038984/208670996-e516e934-7e28-4156-adad-5d7bfbb8c5e2.png)

Basically, the point is that people will prefer a direct train with no transfers than a train with transfers. At the moment, people dont want to take a slow passenger train because it stops one more time on the way, so they prefer to wait 30 minutes for the next train.

**Comment by luca009** [comment]
I think time-aware cargo distribution would be an awesome feature! The issue is that it's probably pretty hard to implement. In the meantime, I've resorted to turning down my "Saturation of short paths before using high-capacity paths" setting down to something like 20%, which yields some better results than the default and makes your express trains a little less overcrowded, as the people will prefer the less crowded stopper. I would love the passengers to follow the schedule, like you suggested, though.
